src/cli.rs

```diff
@@ -64,6 +69,121 @@ fn check_path(
     })
 }
 
+/// This will generate a clap subcommand ([`App`]) that can be used for operations on CRDs.
+///
+/// Currently two arguments are supported:
+/// * `print`: This will print the schema to stdout
+/// * `save`: This will save the schema to a file
+///
+/// The resulting subcommand can be handled by the [`self::handle_crd_subcommand`] method.
+
+/// # Arguments
+///
+/// * `name`: Name of the CRD
+///
+/// returns: App
+///
+/// # Examples
+///
+/// ```
+/// use kube::CustomResource;
+/// use schemars::JsonSchema;
+/// use serde::{Serialize, Deserialize};
+///
+/// #[derive(Clone, CustomResource, Debug, JsonSchema, Serialize, Deserialize)]
+/// #[kube(
+///     group = "foo.stackable.tech",
+///     version = "v1",
+///     kind = "FooCluster",
+///     namespaced
+/// )]
+/// pub struct FooClusterSpec {
+///     pub name: String,
+/// }
+///
+/// let command = stackable_operator::cli::generate_crd_subcommand::<FooCluster>();
+/// ```
+pub fn generate_crd_subcommand<'a, 'b, T>() -> App<'a, 'b>
+where
+    T: CustomResourceExt,
+{
+    let kind = T::api_resource().kind;
+
+    SubCommand::with_name(&kind.to_lowercase())
+        .arg(
+            Arg::with_name("print")
+                .short("p")
+                .long("print")
+                .help("Will print the CRD schema in YAML format to stdout"),
+        )
+        .arg(
+            Arg::with_name("save")
+                .short("s")
+                .long("save")
+                .takes_value(true)
+                .value_name("FILE")
+                .conflicts_with("print")
+                .help("Will write the CRD schema in YAML format to the specified location"),
+        )
+}
+
+/// This will handle a subcommand generated by the [`self::generate_crd_subcommand`] method.
+///
+/// The CRD and the name of the subcommand will be identified by the `kind` of the generic parameter `T` being passed in.
+///
+/// # Arguments
+///
+/// * `matches`: The [`ArgMatches`] object which _might_ contain a match for our current CRD.
+///
+/// returns: A boolean wrapped in a result indicating whether the this method did handle the argument.
+///          If it returns `Ok(true)` the program should abort.
```

Comment:
Can we  enforce that the program should abort and do `std::process::exit(0)` instead of returning `Ok(true)`?

Reply:
We could probably exit here, but I'm not sure if that is a call that a library should make. 
Only the application using this method can know what the exact repercussions are if this fails and if it needs to abort or can maybe somehow recover..